src/frontend/screens/Security/index.tsx

```diff
@@ -64,52 +65,70 @@ const m = defineMessages({
 });
 
 export const Security: NavigationStackScreenComponent = () => {
-  const { passIsSet } = React.useContext(SecurityContext);
   const [{ experiments }] = React.useContext(SettingsContext);
+  const [authState, setAuthState] = React.useContext(SecurityContext);
+  const [highlight, setHighlight] = React.useState(false);
   const { navigate } = useNavigation();
 
   React.useEffect(() => {
     if (!experiments.appPasscode) navigate("Settings");
   }, [experiments]);
 
+  function highlightError() {
+    setHighlight(true);
+    setTimeout(() => {
+      setHighlight(false);
+    }, 2000);
+  }
+
   const [passCodeDes, killPassCodeDes] = React.useMemo(
     () =>
-      passIsSet
+      !!authState.passcode
         ? [m.passDesriptionPassSet, m.killPassDescriptonPassSet]
         : [m.passDesriptionPassNotSet, m.killPassDescriptonPassNotSet],
-    [passIsSet]
+    [authState.passcode]
   );
 
   return (
     <List>
-      <ListItem button={false} style={{ marginVertical: 10 }}>
-        <ListItemText
-          style={{ textTransform: "uppercase" }}
-          primary={<FormattedMessage {...m.securitySubheader} />}
-        />
-      </ListItem>
-
-      <ListItem button={true} onPress={() => navigate("AppPasscode")}>
-        <ListItemText
-          primary={<FormattedMessage {...m.passcodeHeader} />}
-          secondary={<FormattedMessage {...passCodeDes} />}
-        />
-      </ListItem>
+      {authState.appMode === "normal" && (
+        <>
+          <ListItem button={false} style={{ marginVertical: 10 }}>
+            <ListItemText
+              style={{ textTransform: "uppercase" }}
+              primary={<FormattedMessage {...m.securitySubheader} />}
+            />
+          </ListItem>
 
-      <ListItem
-        button={true}
-        onPress={() => {
-          return;
-        }}
-      >
-        <ListItemText
-          primary={<FormattedMessage {...m.killPasscodeHeader} />}
-          secondary={<FormattedMessage {...killPassCodeDes} />}
-        />
-      </ListItem>
+          <ListItem button={true} onPress={() => navigate("AppPasscode")}>
+            <ListItemText
+              primary={<FormattedMessage {...m.passcodeHeader} />}
+              secondary={<FormattedMessage {...passCodeDes} />}
+            />
+          </ListItem>
 
-      <ListDivider style={styles.divder} />
+          <ListItem
+            onPress={() => {
+              if (!authState.passcode) {
+                highlightError();
+                return;
+              }
+              navigate("KillPasscode");
+            }}
+          >
```

Comment:
not a blocker but feel like it would be more useful to disable the option to enable this in the actual kill passcode screen, rather in this security screen. in its current state, it's impossible to even find out what the kill passcode is, which i feel could be useful for someone who's considering the option but hasn't commited to creating an app passcode just yet

Reply:
Yeah, perhaps we should bring this up with Sabella, because this is her current design flow, but I do agree with your sentiment.